CLI/source-git/init: update after the CLI overhaul (#330)
packit/packit.dev: content/docs/CLI/source-git/init.md

---

> title: "packit source-git init"

Reviewer: The title is shown in the left side menu and with your change, we'd see

```
User Documentation
|- Packit CLI
   |- source-git
      |- packit source-git init
```
which is a bit redundant

Author: ah, so I revert it back to just init

draft: false
disableToc: false
weight: 1
---

> # `packit source-git init`

Reviewer: I'm not against it, but I prefer consistency so if you want this, please do the same change on all pages.

Author: oki, that's a good point, I'll do that

## Prerequisites

 * A clone of the upstream repo with checked out branch which matches the base ref.

 * Git ref of the upstream repo used as a base.

 * Clean dist-git repository.

## Tutorial

As an example, let's create a source-git repo out of
[systemd](https://github.com/systemd/systemd-stable). The project uses a
dedicated repository to provide backports for actively maintained stable
releases.

First step is to clone the respective dist-git repo (we'll use Fedora Rawhide
in our case):

    $ cd $FEDORA_DIST_GIT_REPOS
    $ fedpkg clone systemd
    $ cd systemd

We should now determine the version in the specfile

    $ rpmspec -q --qf "%{version}" ./systemd.spec
    249.4

Since systemd-stable prefixes the backport releases with `v`, we know that we want to base our source-git repo on the `v249.4` tag:

    $ cd $FEDORA_SOURCE_GIT_REPOS
    $ git clone https://github.com/systemd/systemd-stable
    $ cd systemd-stable
    $ git checkout 'v249.4'

All the prereqs should be met now! We have the upstream clone with the proper
tag checked out and the respective dist-git repo. We can now run the `init`
command:

>     $ packit init v249.4 $FEDORA_SOURCE_GIT_REPOS/systemd-stable $FEDORA_DIST_GIT_REPOS/systemd

Reviewer: ```suggestion
    $ packit source-git init v249.4 $FEDORA_SOURCE_GIT_REPOS/systemd-stable $FEDORA_DIST_GIT_REPOS/systemd
```

Author: apparently I'm not good at copy-pasting from terminal :D

weight: 1
---

# `packit source-git init`

Initialize a repository to serve as a [source-git repo]({{< ref "/docs/source-git" >}}).

The biggest advantage of this command is the functionality to create source-git
repos. Here's a quick rundown what packit does:

>  * Fetches specfile and additional sources from Fedora or CentOS dist-git.
> 
>  * Apply downstream patches as commits.
> 
>  * Add packit config file.

Reviewer 2: ```suggestion
 * Fetches specfile and additional sources from Fedora or CentOS dist-git.
 * Apply downstream patches as commits.
 * Add source-git configuration to control future syncing to dist-git.
```

We have [a guide]({{< ref "/docs/source-git/how-to-source-git.md" >}}) on how

> to create source-git repos manually if you want to understand precise steps packit performs.

Reviewer 2: ```suggestion
to create source-git repos manually if you want to understand the steps performed.
```

## Prerequisites

>  * A clone of the upstream repo with checked out branch which matches the base ref.
> 
>  * Git ref of the upstream repo used as a base.
> 
>  * Clean dist-git repository.

Reviewer 2: ```suggestion
 * A clone of the upstream repo with checked out branch which matches the base ref.
 * The git ref of the upstream repo used as a base.
 * A clean dist-git repository.
```

## Tutorial

As an example, let's create a source-git repo out of
[systemd](https://github.com/systemd/systemd-stable). The project uses a
dedicated repository to provide backports for actively maintained stable
releases.

First step is to clone the respective dist-git repo (we'll use Fedora Rawhide
in our case):

    $ cd $FEDORA_DIST_GIT_REPOS
    $ fedpkg clone systemd
    $ cd systemd

We should now determine the version in the specfile

    $ rpmspec -q --qf "%{version}" ./systemd.spec
    249.4

Since systemd-stable prefixes the backport releases with `v`, we know that we want to base our source-git repo on the `v249.4` tag:

    $ cd $FEDORA_SOURCE_GIT_REPOS
    $ git clone https://github.com/systemd/systemd-stable
    $ cd systemd-stable
    $ git checkout 'v249.4'

All the prereqs should be met now! We have the upstream clone with the proper

> tag checked out and the respective dist-git repo. We can now run the `init`

Reviewer 2: ```suggestion
tag checked out and the respective dist-git repo. We can now run the `source-git init`
```

command:

    $ packit source-git init v249.4 $FEDORA_SOURCE_GIT_REPOS/systemd-stable $FEDORA_DIST_GIT_REPOS/systemd

Once the command finished successfully, we can create a SRPM and build it in
mock to prove the source-git repo works:

    $ cd FEDORA_SOURCE_GIT_REPOS/systemd-stable

>     $ packit srpm

Reviewer 2: Should this still be the command to validate the new setup? Wouldn't running `packit source-git update-dist-git`, and check that dist-git is not modified, be better?

Author: when the source-git repo is created from the respective dist-git repo fresh, update-dist-git should be a noop (which is also a good test). I will mention both - would that work?

Author: okay, it is actually redundant since we have those steps covered in the work-with-source-git section, so I'll just link there instead
